src/NationalInstruments.Analyzers.Utilities/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.CodeAnalysis;
+
+namespace NationalInstruments.Analyzers.Utilities.Extensions
+{
+    /// <summary>
+    /// Class that contains helpful extensions to <see cref="IMethodSymbol"/>.
+    /// </summary>
+    /// <remarks>
+    /// Microsoft has most of these extensions internally, but they haven't made them public.
+    /// </remarks>
+    public static class IMethodSymbolExtensions
+    {
+        /// <summary>
+        /// Checks if the given method is a Finalizer implementation.
+        /// </summary>
+        /// <param name="method">The method to check</param>
+        public static bool IsFinalizer(this IMethodSymbol method)
```

Comment:
Could add a comment that the source is: https://github.com/dotnet/roslyn-analyzers/blob/846a766f73caa82608db6fee9f2860004298449f/src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs#L114